src/index.ts

```diff
@@ -1,279 +1,164 @@
-	const chain = (await api.rpc.system.chain()).toString()
-	const subFn = subscription == ApiSubscription.Head ? api.rpc.chain.subscribeNewHeads: api.rpc.chain.subscribeFinalizedHeads;
-
-	logger.info(`⛓ Connected to [${ws}] ${chain} [ss58: ${api.registry.chainSS58}] [listening: ${subscription}]`)
-	const unsub = await subFn(async (header) => {
-		logger.debug(`checking block ${header.hash} of ${chain}`);
-		await perHeader(chain, api, header, extendedAccounts, reporters, methodSub)
-	});
-}
-
-async function listAllChains(config: AppConfig, reporters: Reporter[]) {
-	const _ = await Promise.all(config.endpoints.map((e) => listenChain(e, config.listen, config.accounts, config.method_subscription, reporters)));
-	// a rather wacky way to make sure this function never returns.
-	return new Promise(() => {});
-}
-
-async function main() {
-	if (!argv.c) {
-		logger.error('-c or DOT_NOTIF_CONF env variable must specify a config file');
-		process.exit(1);
-	}
+			// Make a pretty html and send it to the matrix channel
+			const message = ReportHTML(report);
 
-	const config: AppConfig = JSON.parse(readFileSync(argv.c).toString());
-	const missingField = missingAppConfig(config);
-	if (missingField) {
-		logger.error(`missing some key in config file: ${missingField}`);
-		process.exit(1);
-	}
-
-	if (config.listen !== ApiSubscription.Finalized && config.listen !== ApiSubscription.Head) {
-		logger.warn(`"listen" config not provided or invalid (${config.listen}), overwriting to ${ApiSubscription.Finalized}`);
-		config.listen = ApiSubscription.Finalized;
-	}
-
-	const reporters: Reporter[] = [];
-	for (const reporterType in config.reporters) {
-		if (reporterType === "email") {
-			const reporter = new EmailReporter(config.reporters[reporterType] as EmailConfig);
-			await reporter.verify();
-			reporters.push(reporter)
-		}
-		if (reporterType === "console") {
-			reporters.push(new ConsoleReporter())
-		}
-		if (reporterType === "fs") {
-			reporters.push(new FileSystemReporter(config.reporters[reporterType] as FsConfig))
-		}
-		if (reporterType === "matrix") {
-			const reporter = new MatrixReporter(config.reporters[reporterType] as MatrixConfig);
-			reporters.push(reporter)
-		}
-	}
-
-	if (config.accounts.length) {
-		config.accounts.forEach(([address, nick]) => logger.info(`📇 registering address ${address} aka ${nick}.`))
-	} else {
-		logger.info(`⚠️ no list of accounts provided, this will match with anything.`);
-	}
-	logger.info(`👂 method subscription: ${JSON.stringify(config.method_subscription)}`);
-
-	const retry = true;
-	while (retry) {
-		try {
-			await listAllChains(config, reporters);
-			// this is just to make sure we NEVER reach this code. The above function never returns,
-			// unless if we trap in an exception.
-			process.exit(1);
-		} catch (e) {
-			logger.error(`retrying due to error: ${e}`)
+			const subject = `${report.chain} notification at ${report.blocknumber}`;
```

Comment:
you are not leveraging the OOP aspect of reports here, the fact that they all implement `Reporter` and you can just have an array of reporters, and once you want to report anything, you simply call `await Promise.all(reporters.map((r) => r.report(report)))`

removing this abstraction is actually highly against simplicity, I highly recommend bringing it back.